From 1818d0780405ec749a487ebec7c528e860853831 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 12 Jul 2018 11:53:55 +0300
Subject: [PATCH 4155/5242] MLK-18877-2: arm64: dts: imx8mq: Add dts bindings
 for adv7535 i2c int

commit  ed6fb195b1fe8810a80e97791f51cc7cf6c8e3cc from
https://source.codeaurora.org/external/imx/linux-imx.git

Add dts bindings for the adv7535 i2c interrupts on imx8mq. These
interrupts are used to catch the HPD event from HDMI sink.
Also, rise the max-res of the lcdif to 1920x1080, when used as
standalone. Since the DCSS is turned off on this dts file, the LCDIF can
be used at full-hd resolution.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts  |   15 +++++++++++++++
 .../dts/freescale/fsl-imx8mq-evk-dual-display.dts  |   15 +++++++++++++++
 .../dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts |   14 ++++++++++++++
 3 files changed, 44 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
index afe9268..d8f2575 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
@@ -56,6 +56,10 @@
 		reg = <0x3d>;
 		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
+		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+
 		status = "okay";
 
 		port {
@@ -90,3 +94,14 @@
 		};
 	};
 };
+
+&iomuxc {
+	imx8mq-evk {
+		pinctrl_i2c1_dsi_ts_int: dsi_ts_int {
+			fsl,pins = <
+				MX8MQ_IOMUXC_ECSPI1_MOSI_GPIO5_IO7              0x19
+			>;
+		};
+
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
index 2c51269..c62dba2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
@@ -30,6 +30,10 @@
 		reg = <0x3d>;
 		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
+		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+
 		status = "okay";
 
 		port {
@@ -66,3 +70,14 @@
 		};
 	};
 };
+
+&iomuxc {
+	imx8mq-evk {
+		pinctrl_i2c1_dsi_ts_int: dsi_ts_int {
+			fsl,pins = <
+				MX8MQ_IOMUXC_ECSPI1_MOSI_GPIO5_IO7              0x19
+			>;
+		};
+
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
index 204d696..54ff0f6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
@@ -34,6 +34,7 @@
 
 &lcdif {
 	status = "okay";
+	max-res = <1920>, <1080>;
 
 	port@0 {
 		lcdif_mipi_dsi: mipi-dsi-endpoint {
@@ -48,6 +49,10 @@
 		reg = <0x3d>;
 		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
+		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+
 		status = "okay";
 
 		port {
@@ -85,3 +90,12 @@
 	};
 };
 
+&iomuxc {
+	imx8mq-evk {
+		pinctrl_i2c1_dsi_ts_int: dsi_ts_int {
+			fsl,pins = <
+				MX8MQ_IOMUXC_ECSPI1_MOSI_GPIO5_IO7              0x19
+			>;
+		};
+	};
+};
-- 
1.7.9.5

